framework/src/media/audio/audio_manager.c

```diff
@@ -117,21 +122,54 @@ audio_manager_result_t find_input_audio_card(void)
 	}
 
 	while ((dir_entry = readdir(dir_info)) != NULL) {
+		char path[256] = "/dev/audio/";
 		auddbg("Found a directory %s\n", dir_entry->d_name);
 		// TODO: Add cases for various drivers. Currently, identify 'pcm' drivers only.
 		if ((dir_entry->d_name[0] != 'p')
 			|| (dir_entry->d_name[1] != 'c')
 			|| (dir_entry->d_name[2] != 'm')
 			|| (sscanf(&dir_entry->d_name[3], "C%uD%u%c", &card_id, &device_id, &type) != 3)) {
-			closedir(dir_info);
-			return AUDIO_MANAGER_INVALID_DEVICE_NAME;
+			continue;
 		}
 
+		strcat(path, dir_entry->d_name);
 		if ((type == 'c') && (card_id < CONFIG_AUDIO_MAX_INPUT_CARD_NUM) && (device_id < MAX_AUDIO_DEVICE_NUM)) {
+#ifndef CONFIG_AUDIO_EXCLUDE_VOLUME
+			int fd;
+			struct audio_caps_desc_s caps_desc;
+			FAR struct audio_caps_s *caps = &(caps_desc.caps);
+
+			if ((fd = open(path, O_RDONLY)) < 0) {
+				auddbg("Fail to open input driver, path = %s\n", path);
+				return AUDIO_MANAGER_DEVICE_FAIL;
+			}
+
+			caps->ac_len = sizeof(struct audio_caps_s);
+			caps->ac_type = AUDIO_TYPE_FEATURE;
+			caps->ac_format.hw = AUDIO_FU_INP_GAIN;
+
+			if ((ret = ioctl(fd, AUDIOIOC_GETCAPS, (unsigned long)caps)) < 0) {
+				auddbg("Fail to ioctl AUDIOIOC_GETCAPS, ret = %d\n", ret);
+				close(fd);
```

Comment:
ret = AUDIO_MANAGER_DEVICE_FAIL;
goto errour_with_dir

Reply:
'error_out' and 'goto' are used now.